### ISSUE TITLE
Is there any way to avoid boring cast or templating ?

### ISSUE THREAD

**Issue by SilNak** [tpierrain/NFluent#135]
Hi folks !

I'd like to speek nfluently when I'm writing such assertion:
```
            ushort usValue = 0;
            Check.That(usValue).IsEqualTo(0);
```

// Oops this fails :
with a gentle message :

> NFluent.FluentCheckException: 
> The checked value is different from the expected one.
> The checked value:
> 	[0] of type: [ushort]
> The expected value:
> 	[0] of type: [int]

Well, .Net compiler consider '0' number as an integer by default.

So to get around this I'm obliged to cast or template that way:

```
            ushort usValue = 0;
            Check.That(usValue).IsEqualTo((ushort)0);

```
or
```
            ushort usValue = 0;
            Check.That(usValue).IsEqualTo<ushort>(0);
```
This is more verbose, any way to avoid this ?

**Comment by dupdob** [comment]
Thanks, issue has been reproduced.

**Comment by dupdob** [comment]
I came to the realisation that this issue will be *hard to fix

**Comment by dupdob** [comment]
Closes it is a duplicate of #138

**Comment by dupdob** [comment]
The issue came from the resolution rules for extension methods. A **fix** will be available in my repo shortly.
